compiler/src/test/java/com/squareup/anvil/compiler/codegen/ContributesSubcomponentHandlerGeneratorTest.kt

```diff
@@ -1296,7 +1297,13 @@ class ContributesSubcomponentHandlerGeneratorTest {
     val packageSuffix = generateSequence(parent) { it.enclosingClass }
       .toList()
       .reversed()
-      .joinToString(separator = ".") { it.simpleName }
+      .joinToString(separator = "") {
+        val className = it.simpleName
+        className
+          .filterIndexed { index, c ->
+            index == 0 || c.isUpperCase() || c.isDigit() || className[index - 1].isUpperCase()
+          }
+      }
```

Comment:
This feels like something that might be worth making a more visible utility method so both the source file and test file can reuse it, and then having a separate test for the utility to catch when that behavior changes.